src/common/decorators/local-storage.ts

```diff
@@ -0,0 +1,64 @@
+import type { ClassElement } from "../../types";
+
+class Storage {
+  private _storage: any = {};
+
+  public addFromStorage(storageKey: any): void {
+    if (!this._storage[storageKey]) {
+      const data = window.localStorage.getItem(storageKey);
+      if (data) {
+        this._storage[storageKey] = JSON.parse(data);
+      }
+    }
+  }
+
+  public hasKey(storageKey: string): any {
+    return storageKey in this._storage;
+  }
+
+  public getValue(storageKey: string): any {
+    return this._storage[storageKey];
+  }
+
+  public setValue(storageKey: string, value: any): any {
+    this._storage[storageKey] = value;
+    window.localStorage.setItem(storageKey, JSON.stringify(value));
```

Comment:
This raises in Safari private mode.

Reply:
I also think that there were some cases where it was just undefined

Reply:
`localStorage` ?